crates/bevy_reflect/src/struct_trait.rs

```diff
@@ -60,6 +63,71 @@ pub trait Struct: Reflect {
     fn clone_dynamic(&self) -> DynamicStruct;
 }
 
+/// A container for compile-time struct info
+#[derive(Clone, Debug)]
+pub struct StructInfo {
+    id: TypeIdentity,
+    fields: Box<[NamedField]>,
+    field_indices: HashMap<Cow<'static, str>, usize>,
+}
+
+impl StructInfo {
+    /// Create a new [`StructInfo`]
+    ///
+    /// # Arguments
+    ///
+    /// * `fields`: The fields of this struct in the order they are defined
+    ///
+    pub fn new<T: Reflect>(fields: &[NamedField]) -> Self {
+        let field_indices = fields
+            .iter()
+            .enumerate()
+            .map(|(index, field)| {
+                let name = field.name().to_string();
+                (Cow::Owned(name), index)
```

Comment:
If the name here is  Cow::Borrowed, you could probably clone it without incurring an allocation here.

Reply:
It's saying that in order to do this, I need to make `fields` static, which I don't think I can do. This might also be because I'm not sure how to make `field.name()` return `&'static str` instead of just `&str`.

Reply:
Can you do

```rust
fn name(&'static self) -> &'static str {
}
```

Reply:
> Can you do
> 
> ```rust
> fn name(&'static self) -> &'static str {
> }
> ```

Looks like I can, but then it requires that `fields` be static too 😭

Reply:
> It's saying that in order to do this, I need to make `fields` static, which I don't think I can do. This might also be because I'm not sure how to make `field.name()` return `&'static str` instead of just `&str`.

Type name can return `&Cow<'static, str>`,  which is `Clone`.

Reply:
That works! And I think I now understand this `Cow` thing a little more haha